ui/src/menu.test.tsx

```diff
@@ -0,0 +1,36 @@
+import Button from "@mui/material/Button";
+import MenuItem from "@mui/material/MenuItem";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import React from "react";
+import { useMenu } from "./menu";
+
+function MenuButton(props: { onClick: () => void }) {
+  const [menu, show] = useMenu({
+    children: [
+      <MenuItem key="1" onClick={props.onClick}>
+        menu-item
+      </MenuItem>,
+    ],
+  });
+  return (
+    <>
+      <Button onClick={show}>show</Button>
+      {menu}
+    </>
+  );
+}
+
+test("show/hide menu", async () => {
+  const onClick = jest.fn();
+  render(<MenuButton onClick={onClick} />);
+
+  userEvent.click(screen.getByText("show"));
```

Comment:
screen.getByRole("button") or screen.getByRole("button", {name: "show"}) preferred?

ie docs say to prefer getByRole over getByText when possible.

Reply:
Done